src/cloud_provider/aws/kubernetes/mod.rs

```diff
@@ -547,6 +560,112 @@ impl EKS {
         Ok(context)
     }
 
+    /// Returns a tuple of (update_desired_node: bool, desired_nodes_count: i32).
+    fn should_update_desired_nodes(&self, event_details: EventDetails) -> Result<(bool, i32), EngineError> {
+        let future_node_group = match self.nodes_groups.is_empty() {
+            false => self.nodes_groups.first().unwrap(),
```

Comment:
Please avoid unwrap in prod, use expect instead, will be easier to debug (even if in this case it should never be a problem)